### ISSUE TITLE
Problem running a build with new version of cpython (1.6.0) and pip (0.16.0) buildpacks

### ISSUE THREAD

**Issue by sfzylad** [paketo-buildpacks/pip#342]
Version 0.16.0 of `pip` buildpack fails installing pip:

```
[builder] Running build for buildpack paketo-buildpacks/pip@0.16.0
[builder] Looking up buildpack
[builder] Finding plan
[builder] Running build for buildpack Paketo Buildpack for Pip 0.16.0
[builder] Creating plan directory
[builder] Preparing paths
[builder] Running build command
[builder] Paketo Buildpack for Pip 0.16.0
[builder]   Resolving Pip version
[builder]     Candidate version sources (in priority order):
[builder]       BP_PIP_VERSION -> "22.3.0"
[builder]       <unknown>      -> ""
[builder]
[builder]     Selected Pip version (using BP_PIP_VERSION): 22.3.0
[builder]
[builder]   Executing build process
[builder]     Installing Pip 22.3.0
[builder] failed to configure pip:
[builder] Looking in links: /tmp/pip-source2821721444
[builder] Processing /tmp/pip-source2821721444
[builder]   Installing build dependencies: started
[builder]   Installing build dependencies: finished with status 'done'
[builder]   Getting requirements to build wheel: started
[builder]   Getting requirements to build wheel: finished with status 'done'
[builder]   Preparing metadata (pyproject.toml): started
[builder]   Preparing metadata (pyproject.toml): finished with status 'error'
[builder]   error: subprocess-exited-with-error
[builder]
[builder]   × Preparing metadata (pyproject.toml) did not run successfully.
[builder]   │ exit code: 1
[builder]   ╰─> [82 lines of output]
[builder]       /tmp/pip-build-env-l9nbmere/overlay/lib/python3.10/site-packages/setuptools/config/setupcfg.py:508: SetuptoolsDeprecationWarning: The license_file parameter is deprecated, use license_files instead.
[builder]         warnings.warn(msg, warning_class)
[builder]       running dist_info
[builder]       creating /tmp/pip-modern-metadata-3f_ps_la/pip.egg-info
[builder]       writing /tmp/pip-modern-metadata-3f_ps_la/pip.egg-info/PKG-INFO
[builder]       writing dependency_links to /tmp/pip-modern-metadata-3f_ps_la/pip.egg-info/dependency_links.txt
[builder]       writing entry points to /tmp/pip-modern-metadata-3f_ps_la/pip.egg-info/entry_points.txt
[builder]       writing top-level names to /tmp/pip-modern-metadata-3f_ps_la/pip.egg-info/top_level.txt
[builder]       writing manifest file '/tmp/pip-modern-metadata-3f_ps_la/pip.egg-info/SOURCES.txt'
[builder]       reading manifest file '/tmp/pip-modern-metadata-3f_ps_la/pip.egg-info/SOURCES.txt'
[builder]       reading manifest template 'MANIFEST.in'
[builder]       warning: no files found matching 'src/pip/_vendor/pyparsing/diagram/template.jinja2'
[builder]       warning: no files found matching 'docs/docutils.conf'
[builder]       warning: no previously-included files found matching '.coveragerc'
[builder]       warning: no previously-included files found matching '.mailmap'
[builder]       warning: no previously-included files found matching '.appveyor.yml'
[builder]       warning: no previously-included files found matching '.readthedocs.yml'
[builder]       warning: no previously-included files found matching '.pre-commit-config.yaml'
[builder]       warning: no previously-included files found matching 'tox.ini'
[builder]       warning: no previously-included files found matching 'noxfile.py'
[builder]       warning: no files found matching '*.css' under directory 'docs'
[builder]       warning: no previously-included files found matching 'src/pip/_vendor/six'
[builder]       warning: no previously-included files found matching 'src/pip/_vendor/six/moves'
[builder]       warning: no previously-included files matching '*.pyi' found under directory 'src/pip/_vendor'
[builder]       no previously-included directories found matching '.github'
[builder]       no previously-included directories found matching 'docs/build'
[builder]       no previously-included directories found matching 'news'
[builder]       no previously-included directories found matching 'tasks'
[builder]       no previously-included directories found matching 'tests'
[builder]       no previously-included directories found matching 'tools'
[builder]       adding license file 'LICENSE.txt'
[builder]       writing manifest file '/tmp/pip-modern-metadata-3f_ps_la/pip.egg-info/SOURCES.txt'
[builder]       creating '/tmp/pip-modern-metadata-3f_ps_la/pip-22.3.dist-info'
[builder]       Traceback (most recent call last):
[builder]         File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 363, in <module>
[builder]           main()
[builder]         File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 345, in main
[builder]           json_out['return_val'] = hook(**hook_input['kwargs'])
[builder]         File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 164, in prepare_metadata_for_build_wheel
[builder]           return hook(metadata_directory, config_settings)
[builder]         File "/tmp/pip-build-env-l9nbmere/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 377, in prepare_metadata_for_build_wheel
[builder]           self.run_setup()
[builder]         File "/tmp/pip-build-env-l9nbmere/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 335, in run_setup
[builder]           exec(code, locals())
[builder]         File "<string>", line 26, in <module>
[builder]         File "/tmp/pip-build-env-l9nbmere/overlay/lib/python3.10/site-packages/setuptools/__init__.py", line 87, in setup
[builder]           return distutils.core.setup(**attrs)
[builder]         File "/tmp/pip-build-env-l9nbmere/overlay/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 185, in setup
[builder]           return run_commands(dist)
[builder]         File "/tmp/pip-build-env-l9nbmere/overlay/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
[builder]           dist.run_commands()
[builder]         File "/tmp/pip-build-env-l9nbmere/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 968, in run_commands
[builder]           self.run_command(cmd)
[builder]         File "/tmp/pip-build-env-l9nbmere/overlay/lib/python3.10/site-packages/setuptools/dist.py", line 1217, in run_command
[builder]           super().run_command(command)
[builder]         File "/tmp/pip-build-env-l9nbmere/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 987, in run_command
[builder]           cmd_obj.run()
[builder]         File "/tmp/pip-build-env-l9nbmere/overlay/lib/python3.10/site-packages/setuptools/command/dist_info.py", line 101, in run
[builder]           bdist_wheel = self.get_finalized_command('bdist_wheel')
[builder]         File "/tmp/pip-build-env-l9nbmere/overlay/lib/python3.10/site-packages/setuptools/_distutils/cmd.py", line 305, in get_finalized_command
[builder]           cmd_obj = self.distribution.get_command_obj(command, create)
[builder]         File "/tmp/pip-build-env-l9nbmere/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 859, in get_command_obj
[builder]           klass = self.get_command_class(command)
[builder]         File "/tmp/pip-build-env-l9nbmere/overlay/lib/python3.10/site-packages/setuptools/dist.py", line 954, in get_command_class
[builder]           self.cmdclass[command] = cmdclass = ep.load()
[builder]         File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.10/importlib/metadata/__init__.py", line 171, in load
[builder]           module = import_module(match.group('module'))
[builder]         File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.10/importlib/__init__.py", line 126, in import_module
[builder]           return _bootstrap._gcd_import(name[level:], package, level)
[builder]         File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
[builder]         File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
[builder]         File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
[builder]         File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
[builder]         File "<frozen importlib._bootstrap_external>", line 883, in exec_module
[builder]         File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
[builder]         File "/tmp/pip-build-env-l9nbmere/overlay/lib/python3.10/site-packages/wheel/bdist_wheel.py", line 28, in <module>
[builder]           from .macosx_libfile import calculate_macosx_platform_tag
[builder]         File "/tmp/pip-build-env-l9nbmere/overlay/lib/python3.10/site-packages/wheel/macosx_libfile.py", line 43, in <module>
[builder]           import ctypes
[builder]         File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.10/ctypes/__init__.py", line 8, in <module>
[builder]           from _ctypes import Union, Structure, Array
[builder]       ModuleNotFoundError: No module named '_ctypes'
[builder]       [end of output]
[builder]
[builder]   note: This error originates from a subprocess, and is likely not a problem with pip.
[builder] error: metadata-generation-failed
[builder]
[builder] × Encountered error while generating package metadata.
[builder] ╰─> See above for output.
[builder]
[builder] note: This is an issue with the package mentioned above, not pip.
[builder] hint: See above for details.
[builder]
[builder] [notice] A new release of pip available: 22.2.2 -> 22.3.1
[builder] [notice] To update, run: pip3 install --upgrade pip
```

`pack report` output:
```
Pack:
  Version:  0.27.0+git-f4f5be1.build-3382
  OS/Arch:  darwin/amd64

Default Lifecycle Version:  0.14.1

Supported Platform APIs:  0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9

Config:
  default-builder-image = "[REDACTED]"

  [[trusted-builders]]
    name = "[REDACTED]"
```

Problem occurs either with pip 22.3.0 and 22.3.1.

## Expected Behavior

The pip buildpack runs without a problem and the builds moves to another layer of the build.

## Current Behavior

The build fails in pip installation layer.

## Steps to Reproduce

Run build where both cpython v1.6.0 and pip 0.16.0 buildpacks would participate.

**Comment by robdimsdale** [comment]
Hey @sfzylad - yes, we have just observed this too. We are actively investigating. We think it's related to the changes we made to compile `cpython`.

For now, you should be able to use the previous version of `cpython` buildpack while we work on a fix.

**Comment by robdimsdale** [comment]
An update @sfzylad - we identified that we were missing a library when compiling python (specifically `libffi-dev`). We are recompiling python with this library and will release an updated version of cpython with these new dependencies.

Once that release is available, I'll update this issue.

**Comment by robdimsdale** [comment]
@sfzylad this should be fixed with CPython Buildpack [v1.7.1](https://github.com/paketo-buildpacks/cpython/releases/tag/v1.7.1) (or later). The version of `pip` and `pip-install` do not matter as this was an issue with how we compiled CPython.

I'll close this issue now, but feel free to re-open if you continue to see this issue.

**Comment by metacoma** [comment]
hello, i'm encountered this "bug" and the same traceback for https://github.com/paketo-buildpacks/samples/tree/main/python/poetry

steps to reproduce
```
$ pack --version
0.34.2
$ pack config default-builder paketobuildpacks/builder-jammy-tiny
$ git clone github.com/paketo-buildpacks/samples
$ cd samples/python/poetry
# ce3575c68d41a57a2e77638a84d53e39d6cbafa1 commit
$ pack build poetry-sample --buildpack paketo-buildpacks/python
....
             File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.12/importlib/__init__.py", line 90, in import_module
                return _bootstrap._gcd_import(name[level:], package, level)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
              File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
              File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
              File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
              File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
              File "<frozen importlib._bootstrap_external>", line 995, in exec_module
              File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
              File "/tmp/pip-build-env-ftv1aboi/normal/lib/python3.12/site-packages/wheel/bdist_wheel.py", line 27, in <module>
                from .macosx_libfile import calculate_macosx_platform_tag
              File "/tmp/pip-build-env-ftv1aboi/normal/lib/python3.12/site-packages/wheel/macosx_libfile.py", line 43, in <module>
                import ctypes
              File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.12/ctypes/__init__.py", line 8, in <module>
                from _ctypes import Union, Structure, Array
            ModuleNotFoundError: No module named '_ctypes'
            [end of output]
      
        note: This error originates from a subprocess, and is likely not a problem with pip.
      error: metadata-generation-failed
      
      × Encountered error while generating package metadata.
      ╰─> See above for output.
      
      note: This is an issue with the package mentioned above, not pip.
      hint: See above for details.
      [end of output]

.... 
```
